models/users.js

```diff
@@ -58,10 +58,17 @@ const addOrUpdate = async (userData, userId = null) => {
  */
 const fetchUsers = async (query) => {
   try {
-    const snapshot = await userModel
+    let snapshot = await userModel
       .limit(parseInt(query.size) || 100)
       .offset((parseInt(query.size) || 100) * (parseInt(query.page) || 0))
       .get();
+    if (query.role === "members") {
+      snapshot = await userModel
+        .limit(parseInt(query.size) || 100)
+        .offset((parseInt(query.size) || 100) * (parseInt(query.page) || 0))
+        .where("isMember", "==", true)
+        .get();
+    }
```

Comment:
There seem to be repeated lines of code. We can avoid it by calling the `get()` method at the end
```suggestion
    let query = userModel
    query = query.limit(parseInt(query.size) || 100)
      .offset((parseInt(query.size) || 100) * (parseInt(query.page) || 0))
      .get();
    if (query.role === "members") {
      query= query.where("isMember", "==", true);
    }
    const snapshot = await query.get();
```

Reply:
Also, we can make the API more generic by allowing the client to pass any valid role and we can return the users with that role. The client can also pass a list of comma-separated roles and we return all users with those roles.

Reply:
Please avoid using string literal "members" `query.role === "members"`, instead import roles from the [roles.js](https://github.com/Real-Dev-Squad/website-backend/blob/develop/constants/roles.js) file

Reply:
Done